scripts/extensions/pool_update.precheck

```diff
@@ -102,8 +122,20 @@ if __name__ == '__main__':
             update_vdi_uuid = session.xenapi.VDI.get_uuid(update_vdi)
             update_vdi_valid = True
         except Exception as e:
-            print(failure_message(CANNOT_FIND_UPDATE,
-                ['Missing update package [uuid=%s]' % update_uuid]))
+            print(failure_message(CANNOT_FIND_UPDATE, []))
+            sys.exit(0)
+
+        # verify if dom0 has enough space to install the package
+        statvfs = os.statvfs('/')
+        available_dom0_disk_size = statvfs.f_frsize * statvfs.f_bavail
+        try:
+            required_size = int(session.xenapi.pool_update.get_installation_size(update)) * 3
```

Comment:
As packages could be upgraded it is hard to estimate what the actual overhead is. A fixed overhead may not work and a multiplier may be too much. Perhaps a modest multiplier of 1.2?